### PR TITLE
Java: Add lindex command. (List Command Group)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 1]
```diff
@@ -90,18 +90,42 @@ public interface ListBaseCommands {
...
-     * assert payload.equals(new String[] {})
+     * String[] payload = client.lrange("non_exiting_key", 0, 2).get();
+     * assert payload.equals(new String[] {});
      * }</pre>
      */
     CompletableFuture<String[]> lrange(String key, long start, long end);
 
+    /**
+     * Returns the element at <code>index</code> in the list stored at <code>key</code>.<br>
+     * The index is zero-based, so 0 means the first element, 1 the second element and so on. Negative
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:107]: wrap numbers and `null` with `code`

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 2]
```diff
@@ -90,18 +90,42 @@ public interface ListBaseCommands {
...
+     *     If <code>index</code> is out of range or if <code>key</code> does not exist, null is
+     *     returned.
+     * @example
+     *     <pre>{@code
+     * String payload1 = client.lindex("myList", 0).get();
+     * assert payload1.equals('value1'); // Returns the first element in the list stored at 'myList'.
+     *
+     * String payload2 = client.lindex("myList", -1).get();
+     * assert payload2.equals('value3'); // Returns the last element in the list stored at 'myList'.
+     *
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:124]:
```suggestion
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -568,6 +569,27 @@ public T lrange(@NonNull String key, long start, long end) {
         return getThis();
     }
 
+    /**
+     * Returns the element at <code>index</code> in the list stored at <code>key</code>.<br>
+     * The index is zero-based, so 0 means the first element, 1 the second element and so on. Negative
+     * indices can be used to designate elements starting at the tail of the list. Here, -1 means the
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:575]: same here

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 4]
```diff
@@ -583,6 +583,23 @@ public void lpush_lpop_lrange_type_error(BaseClient client) {
...
+        String key = UUID.randomUUID().toString();
+        String[] valueArray =
+                new String[] {
+                    "value" + UUID.randomUUID().toString(), "value" + UUID.randomUUID().toString()
+                };
+
+        assertEquals(2, client.lpush(key, valueArray).get());
+        assertEquals(valueArray[1], client.lindex(key, 0).get());
+        assertEquals(valueArray[0], client.lindex(key, 1).get());
+        assertNull(client.lindex(key, 3).get());
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:599]: add a test where key exists, but it is not a list

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 5]
```diff
@@ -583,6 +583,23 @@ public void lpush_lpop_lrange_type_error(BaseClient client) {
...
     }
 
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void lindex(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String[] valueArray =
+                new String[] {
+                    "value" + UUID.randomUUID().toString(), "value" + UUID.randomUUID().toString()
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:593]: You can use simple `value1` and `value2`

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 6]
```diff
@@ -90,18 +90,42 @@ public interface ListBaseCommands {
...
      * @example
      *     <pre>{@code
-     * String[] payload = lient.lrange("my_list", 0, 2).get()
-     * assert payload.equals(new String[] {"value1", "value2", "value3"})
+     * String[] payload = lient.lrange("my_list", 0, 2).get();
+     * assert payload.equals(new String[] {"value1", "value2", "value3"});
      *
-     * String[] payload = client.lrange("my_list", -2, -1).get()
-     * assert payload.equals(new String[] {"value2", "value3"})
+     * String[] payload = client.lrange("my_list", -2, -1).get();
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:96]: please revert this - I add this to #143

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java (hunk 1 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/ListBaseCommands.java:107]: I don't think it makes sense to wrap the numbers here they just tell about indexing.

Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/ListBaseCommands.java:107]: It will be consistend with other commands, see `ttl` for example

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 7]
```diff
@@ -583,6 +583,26 @@ public void lpush_lpop_lrange_type_error(BaseClient client) {
...
+    public void lindex(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        String[] valueArray = new String[] {"value1", "value2"};
+
+        assertEquals(2, client.lpush(key1, valueArray).get());
+        assertEquals(valueArray[1], client.lindex(key1, 0).get());
+        assertEquals(valueArray[0], client.lindex(key1, 1).get());
+        assertNull(client.lindex(key1, 3).get());
+        assertNull(client.lindex("nonExistingKey", 3).get());
```
Comment (jonathanl-bq) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:598]: Should there be a test for a negative index?

**Review by jonathanl-bq — APPROVED** [review]
Left one question, but otherwise seems good to me.

**Pull request merged**